posts/why-is-react-native-so-promising.md

```diff
@@ -0,0 +1,161 @@
+
+## It spits out native code, but runs in a single thread.
+
+One of the big advantages of using React Native is that, while the core app runs in a JavaScript Engine, it uses JSX to render native code. It's not an application that displays HTML5 or a responsive web application or a hybrid app.
+
+And that means multiple things:
+* The look and feel of the UI is great. No pixels, every corner and border is smooth, Animations are flowing. You actually get 60fps. It's just great.
+* It'll perform as well as a native application (at least on iOS). If you want more in-depth information about it, you should read the article ["Comparing the Performance between Native iOS (Swift) and React-Native"](https://medium.com/the-react-native-log/comparing-the-performance-between-native-ios-swift-and-react-native-7b5490d363e2) by John A. Calderaio
+
+There is however one thing to consider when developing a React Native application. Because Javascript is a single threaded environment, you might want to limit your business code (as you'd do for any front-end app, [SoC](https://en.wikipedia.org/wiki/Separation_of_concerns ) for the win) or you'll have a "freeze" effect on your screen. One solution to this is to use one of ES6's best features, Promises. But even then, this is not exactly another thread being executed. Alexander Zlatkov explains it very well in his ["How JavaScript works"](https://blog.sessionstack.com/how-javascript-works-event-loop-and-the-rise-of-async-programming-5-ways-to-better-coding-with-2f077c4438b5) article.
+
+## React Native is a lot like React.
+
+As mentioned previously, in terms of "developer look and feel" React Native and react are very similar. You use classes, ES6, states and the like, with a few notable differences:
+* The atomic JSX elements won't be the same: No HTML tags here, you'll have to use placeholders. `<View>` is the new `<div>`, `<Text>` is the new `<p>`, `<Image>` is the new `<img>`... and then you have other specific components to create buttons, lists, etc. 
+* Because it's a native application you have to manage the navigation. Basic links won't do. The [documentation](https://facebook.github.io/react-native/docs/navigation.html) explains it very well.
+
+### Redux works with React Native
+
+Not much to say here except that Redux, the React developers' favourite state management library, is also working with React Native with absolutely no difference in the way you'll dev.
+    
+### You can do stateless components and use HOC in React Native
+
+Stateless components and High Order Components (HOC) are, for me, the revelation of the year. It allows a greater flexibility in your component creation and is way lighter than the use of React Component class inheritance. Coming from a backend world, I started React by using classes because that's what I'm used to (Java, guys, Java) and I didn't knew much about functional programming. It took a senior front-end dev to open my eyes on this wonder.
```

Comment:
"what I'm used to" -> "what I'm used to using"
"Java, guys, Java" -> not sure what you mean!
"I didn't knew" -> "I didn't know"

Reply:
I changed the sentence a bit to send the message differently